xtask/src/qemu.rs

```diff
@@ -425,24 +448,37 @@ fn build_esp_dir(opt: &QemuOpt) -> Result<PathBuf> {
         .join(opt.target.as_triple())
         .join(build_mode);
     let esp_dir = build_dir.join("esp");
+
+    // Create boot dir.
     let boot_dir = esp_dir.join("EFI").join("Boot");
-    let built_file = if let Some(example) = &opt.example {
-        build_dir.join("examples").join(format!("{example}.efi"))
-    } else {
-        build_dir.join("uefi-test-runner.efi")
-    };
-    let output_file = match *opt.target {
+    if !boot_dir.exists() {
+        fs_err::create_dir_all(&boot_dir)?;
+    }
+
+    let boot_file_name = match *opt.target {
         UefiArch::AArch64 => "BootAA64.efi",
         UefiArch::IA32 => "BootIA32.efi",
         UefiArch::X86_64 => "BootX64.efi",
     };
-    if !boot_dir.exists() {
-        fs_err::create_dir_all(&boot_dir)?;
-    }
-    fs_err::copy(built_file, boot_dir.join(output_file))?;
 
-    // Add a test file that is used in the media protocol tests.
-    fs_err::write(boot_dir.join("test_input.txt"), "test input data")?;
+    if let Some(example) = &opt.example {
+        let src_path = build_dir.join("examples").join(format!("{example}.efi"));
+        fs_err::copy(src_path, boot_dir.join(boot_file_name))?;
+    } else {
+        let shell_launcher = build_dir.join("shell_launcher.efi");
+        fs_err::copy(shell_launcher, boot_dir.join(boot_file_name))?;
+
+        let test_runner = build_dir.join("uefi-test-runner.efi");
+        fs_err::copy(test_runner, boot_dir.join("test_runner.efi"))?;
+
+        // TODO:
+        // Does Ubuntu (as used in the CI pipeline) even ship the shell app?
```

Comment:
btw: Why don't we ship static OVMF files? They can always be used as fall back.

Reply:
That's a good thought, I've been pondering something like that as well.

Reply:
IIRC from early work on the Shell protocol, OVMF and/or QEMU do not ship an image that has the Shell app. It's a few lines in a build file to add it, however.

Reply:
The shell is actually still present, just not as a separate app. The code in this PR detects that case and boots into the shell by modifing `BootNext`: https://github.com/rust-osdev/uefi-rs/pull/793/files#diff-999f2ef6b67a300d614ceb6ce1c1e7f5c4d7e7ca7e6d2545570fafe499d190fdR43

But, booting the shell app is certainly simpler and once https://github.com/rust-osdev/uefi-rs/pull/798 is done we can just do that.